### PR TITLE
refactor: image metadata facade to dqlite

### PR DESCRIPTION
**Pull request by tlm**
This commit extends work done in the image metadata manager api facade to facilitate a move over to dqlite. In this commit we have removed references to mongo state for the controller uuid and also for model metadata.
    
This now leaves just the image metadata still using mongo based state.
## Checklist

- [x] Code style: imports ordered, good names, simple structure, etc
- [x] Comments saying why design decisions were made
- [x] Go unit tests, with comments saying what you're testing
- ~[ ] [Integration tests](https://github.com/juju/juju/tree/main/tests), with comments saying what you're testing~
- ~[ ] [doc.go](https://discourse.charmhub.io/t/readme-in-packages/451) added or updated in changed packages~

## QA steps

Unit tests in affected package are sufficient for this PR. We are not changing any logic just the source of information.

## Documentation changes

N/A

## Links

**Jira card:** JUJU-6427

### CODE REVIEW

File: apiserver/facades/client/imagemetadatamanager/metadata_test.go [hunk 1]
```diff
@@ -111,26 +108,34 @@ func (s *metadataSuite) TestFindOrder(c *gc.C) {
...
 }
 
 func (s *metadataSuite) TestSaveEmpty(c *gc.C) {
-	ctrl, _ := s.setupAPI(c)
-	defer ctrl.Finish()
+	defer s.setupAPI(c).Finish()
+
+	s.modelInfoService.EXPECT().GetModelInfo(gomock.Any()).Return(
+		coremodel.ReadOnlyModel{
+			CloudRegion: "",
```
Comment (barrettj12) [inline apiserver/facades/client/imagemetadatamanager/metadata_test.go:119]: should we have an actual region value here? And not just empty?

File: apiserver/facades/client/imagemetadatamanager/metadata_test.go [hunk 2]
```diff
@@ -190,27 +194,28 @@ func (s *metadataSuite) TestSaveModelCfgFailed(c *gc.C) {
 	}
 
 	msg := "save error"
-	modelConfigService.EXPECT().ModelConfig(gomock.Any()).Return(nil, errors.New(msg))
+	s.modelConfigService.EXPECT().ModelConfig(gomock.Any()).Return(nil, errors.New(msg))
+	s.modelInfoService.EXPECT().GetModelInfo(gomock.Any()).Return(
+		coremodel.ReadOnlyModel{
+			CloudRegion: "",
```
Comment (barrettj12) [inline apiserver/facades/client/imagemetadatamanager/metadata_test.go:200]: ditto

**Review by barrettj12 — APPROVED** [review]

File: apiserver/facades/client/imagemetadatamanager/metadata_test.go (hunk 1 above)
Comment (tlm) [reply apiserver/facades/client/imagemetadatamanager/metadata_test.go]: We should

File: apiserver/facades/client/imagemetadatamanager/metadata_test.go (hunk 2 above)
Comment (tlm) [reply apiserver/facades/client/imagemetadatamanager/metadata_test.go]: Done & Done

**Pull request merged**